MaraBot/Commands/RaceCommandModule.cs

```diff
@@ -8,33 +10,78 @@ namespace MaraBot.Commands
 {
     using Core;
     using Messages;
-    
+
     public class RaceCommandModule : BaseCommandModule
     {
         public IReadOnlyDictionary<string, Preset> Presets { private get; set; }
+        public IReadOnlyDictionary<string, Option> Options { private get; set; }
 
         [Command("race")]
         [Cooldown(15, 900, CooldownBucketType.User)]
         [RequireGuild]
         public async Task Execute(CommandContext ctx, string presetName)
+        {
+            var success = (presetName == "custom") ?
+                LaunchCustomRace(ctx) :
+                LaunchPresetRace(ctx, presetName);
+
+            var emoji = DiscordEmoji.FromName(ctx.Client, await success ? Display.kValidCommandEmoji : Display.kInvalidCommandEmoji);
+            await ctx.Message.CreateReactionAsync(emoji);
+        }
+
+        public async Task<bool> LaunchCustomRace(CommandContext ctx)
+        {
+            if (ctx.Message.Attachments == null || ctx.Message.Attachments.Count == 0)
+            {
+                await ctx.RespondAsync("No attachment for custom race. You must supply a valid json file.");
+                return false;
+            }
+
+            var attachment = ctx.Message.Attachments[0];
+            var url = attachment.Url;
+
+            var request = WebRequest.Create(url);
+            var responseTask = request.GetResponseAsync();
+
+            if (Path.GetExtension(attachment.FileName).ToLower() != ".json")
+            {
+                await ctx.RespondAsync("Invalid json file.");
```

Comment:
Maybe change the text to "File must be a .json file".

Reply:
We're saying file twice in the same sentence though.

How about:
"Expected file extension to be .json."

Reply:
That's a better idea

Reply:
Fixed in 13c423f5dc37853476b5e9724ae159feada0c67d